### PR TITLE
changed getStorageValue to return an optional

### PR DESCRIPTION
**Pull request by aionick**
## Description

Please include a brief summary of the change that this pull request proposes. Include any relevant motivation and context. List any dependencies required for this change.

- Changed IRepositoryQuery so that its getStorageValue return type is an Optional. This forces implementors to handle the 'missing value' case identically, and this consistency is important for some pre-compiled contracts. This PR current has the .gitmodules pointing to the add_optional2db branch in the fastVM. Once  <a href="https://github.com/aionnetwork/aion_fastvm/pull/12">this corresponding PR</a> in the fastVM is merged I will update the .gitmodules file to no longer point to this branch, it's just here in the meantime for Jenkins.

Fixes Issue # .

## Type of change

Insert **x** into the following checkboxes to confirm (eg. [x]):
- [ ] Bug fix.
- [ ] New feature.
- [x] Enhancement.
- [ ] Unit test.
- [ ] Breaking change (a fix or feature that causes existing functionality to not work as expected).
- [ ] Requires documentation update.

## Testing

Please describe the tests you used to validate this pull request. Provide any relevant details for test configurations as well as any instructions to reproduce these results.

- Ran the kernel and tested against the current test suite.

## Verification

Insert **x** into the following checkboxes to confirm (eg. [x]):
- [x] I have self-reviewed my own code and conformed to the style guidelines of this project.
- [x] New and existing tests pass locally with my changes.
- [ ] I have added tests for my fix or feature.
- [x] I have made appropriate changes to the corresponding documentation.
- [x] My code generates no new warnings.
- [x] Any dependent changes have been made.

### CODE REVIEW

File: modAionImpl/test/org/aion/zero/impl/db/AionRepositoryImplTest.java [hunk 1]
```diff
@@ -146,7 +146,7 @@ public void testAccountStateUpdateStorageRow() {
         track.flush();
 
         byte[] retrievedValue =
-                repository.getStorageValue(defaultAccount, new DataWord(key)).getNoLeadZeroesData();
+                repository.getStorageValue(defaultAccount, new DataWord(key)).get().getNoLeadZeroesData();
```
Comment (JayT106) [inline modAionImpl/test/org/aion/zero/impl/db/AionRepositoryImplTest.java]: what if the getStorageValue return Optional.empty() ?

Comment (aionick) [reply modAionImpl/test/org/aion/zero/impl/db/AionRepositoryImplTest.java]: I'll update these tests so that if empty is returned they fail

**Comment by qoire** [comment]
Not sure I see the benefits of adding ``Optional`` into our critical path. Invoking an additional object on every storage query is sure to add some overhead. Why not just document that storageValue can possibly return null?

I understand that ``Optional`` would enforce this relationship, but probably not a luxury we can afford on our critical path.

**Comment by AlexandraRoatis** [comment]
@qoire do you still see it as a problem if the implementation is changed to pass up the object created in the `IKeyValueStore`?

**Pull request closed without merging**